includes/Admin/Cross_Origin_Isolation.php

```diff
@@ -118,40 +101,36 @@ public function admin_footer() {
 	 *
 	 * @param string $html HTML document as string.
 	 *
-	 * @return string
+	 * @return string Processed HTML document.
 	 */
 	protected function replace_in_dom( string $html ): string {
-		$document = Document::fromHtml( $html );
-
-		if ( ! $document ) {
-			return $html;
-		}
-
-		$map = [
-			'link'   => 'href',
-			'img'    => 'src',
-			'iframe' => 'src',
-			'script' => 'src',
+		$tags = [
+			'link',
+			'img',
+			'iframe',
+			'script',
 		];
 
+		$tags      = implode( '|', $tags );
+		$matches   = [];
 		$processed = [];
 
-		foreach ( $map as $tag => $attribute ) {
-			$tags = $document->getElementsByTagName( $tag );
-			foreach ( $tags as $node ) {
-				$value = $node->getAttribute( $attribute );
-				if ( ! $value ) {
+		if ( preg_match_all( '#<(?P<tag>' . $tags . ')[^<]*?(?:>[\s\S]*?</(?P=tag)>|\s*/>)#', $html, $matches ) ) {
+			foreach ( $matches[0] as $match ) {
+				if ( false !== strpos( $match, 'crossorigin=' ) ) {
 					continue;
 				}
 
-				// If already processed tag, attribute and value before, skip.
-				if ( isset( $processed[ $attribute ] ) && in_array( $value, $processed[ $attribute ], true ) ) {
+				$match_value = [];
+				if ( ! preg_match( '/(src|href)=("([^"]+)"|\'([^\']+)\')/', $match, $match_value ) ) {
 					continue;
 				}
 
-				// Check to see if tag already has attirbute.
-				$cross_origin = $node->getAttribute( 'crossorigin' );
-				if ( $cross_origin ) {
+				$attribute = $match_value[1];
+				$value     = $match_value[4] ?? $match_value[3];
+
+				// If already processed tag, attribute and value before, skip.
+				if ( isset( $processed[ $attribute ] ) && in_array( $value, $processed[ $attribute ], true ) ) {
```

Comment:
Was trying avoid regex and preg matches as they are expensive.

Reply:
I know—and now it turns out using `Document` is more expensive. 60 seconds for parsing is unacceptable. 

Waiting for @divyarajmasani to share performance results of this PR so we have a before/after comparison.

Reply:
Just uploaded a screenshot in the PR description above that shows how this change improves the response time significantly. The editor loads 20x faster!